distribution/archives/build.gradle

```diff
@@ -110,34 +112,58 @@ tasks.withType(AbstractArchiveTask).configureEach {
   baseName = "elasticsearch${subdir.contains('oss') ? '-oss' : ''}"
 }
 
+
 Closure commonZipConfig = {
   dirMode 0755
   fileMode 0644
 }
 
+tasks.withType(Copy).configureEach {
+  dependsOn createLogsDir, createPluginsDir, createJvmOptionsDir
+  dirMode 0755
+  fileMode 0644
+  includeEmptyDirs = true
+  into("build/${name}")
+}
+
+tasks.register("installIntegTestZip", Copy) {
```

Comment:
I suspect we could autogenerate all these install tasks by just creating one for each Zip or Tar task using the original "build" task as the copyspec.

Reply:
I realize now that it's more complicated than that, since you can't register a task inside a lazy evaluation callback. We definitely don't want to eagerly realize all these distribution tasks (building these copy specs is actually very expensive). Perhaps we could use a task rule matching anything with an "install" prefix, and create the install tasks there?

Reply:
yeah directly adding these tasks here is just my initial straight forward way to test the general idea. I think those tasks ultimately need to live in a plugin and distribution project basically just declares what's in that copy spec to some degree. Also I think we can and should get rid of the artificial sub projects for each distribution type. I'm exploring what a good migration path would look like though as this needs to be done in a compatible way with the bwcVersion testing.

Reply:
Yeah, one motivation for the artificial subprojects is the ability to build multiple distributions in parallel. That might not be a major motivator though as locally (and even in CI mostly) we really only ever build two distributions (default and OSS for the current platform).

Reply:
> That might not be a major motivator though as locally (and even in CI mostly) we really only ever build two distributions

This is not true when running packaging tests.

Reply:
I do not see how having those per distribution subproject help with parallization. @rjernst can you maybe point me to the packaging tests?

Reply:
The packaging tests exist under `qa/os`. We take every distribution, and run a series of tests against every supported OS (some are noops like linux distributions on windows).

As for parallelization, it looks like this changed at some point, although I don't know when. We used to have the relevant build tasks for each distribution within the subproject for that distro, eg `:distribution:archives:linux-tar`. But it looks like we now have all the archive tasks within `:distribution:archives`, so as I understand it, these cannot be built in parallel. When we run packaging tests, we need all these distributions.

As for this install task, I thought setting up cross project task dependencies was frowned upon? Could we instead split the build tasks into a task to setup an installation in build, then the archiving task becomes simply archiving that directory as is? Then we add project artifacts for these directories, built by the corresponding tasks, in addition to the archives themselves (which are still needed by packaging tests).